bundle-hack/update_bundle.sh

```diff
@@ -1,5 +1,122 @@
+  exit 2
+fi
+
+# Corrected CSV_FILE variable definition with closing parenthesis
+CSV_FILE="${MANIFESTS_DIR:-/manifests}/external-dns-operator.clusterserviceversion.yaml"
+
+# Update direct references in CSV to match release targets
+sed -i -e "s|quay.io/openshift/origin-external-dns-operator:latest|${OPERATOR_PULLSPEC}|g" \
+       -e "s|k8s.gcr.io/external-dns/external-dns:.*$|${OPERAND_PULLSPEC}|g" \
+       -e "s|quay.io/openshift/origin-external-dns:.*$|${OPERAND_PULLSPEC}|g" \
+       -e "s|quay.io/openshift/origin-external-dns@.*$|${OPERAND_PULLSPEC}|g" \
+       -e "s|quay.io/external-dns-operator/external-dns:.*$|${OPERAND_PULLSPEC}|g" \
+       -e "s|quay.io/external-dns-operator/external-dns@.*$|${OPERAND_PULLSPEC}|g" \
+       -e "s|quay.io/openshift/origin-kube-rbac-proxy:latest|registry.redhat.io/openshift4/ose-kube-rbac-proxy:v4.13|g" "${CSV_FILE}"
+
+export EPOC_TIMESTAMP=$(date +%s)
+export TARGET_CSV_FILE="${CSV_FILE}"
+
+# Combined all Python logic into a single block
+python3 - << CSV_UPDATE
+import os
+from sys import exit as sys_exit
+from datetime import datetime
+from ruamel.yaml import YAML
+
+yaml = YAML()
+
+def load_manifest(pathn):
+   if not pathn.endswith(".yaml"):
+      return None
+   try:
+      with open(pathn, "r") as f:
+         return yaml.load(f)
+   except FileNotFoundError:
+      print("File can not be found")
+      exit(6)
+
+def dump_manifest(pathn, manifest):
+   with open(pathn, "w") as f:
+      yaml.dump(manifest, f)
+   return
+
+timestamp = int(os.getenv('EPOC_TIMESTAMP'))
+datetime_time = datetime.fromtimestamp(timestamp)
+version = os.getenv('VERSION')
+replaces = os.getenv('REPLACES_VERSION')
+operator_pullspec = os.getenv('OPERATOR_IMAGE_PULLSPEC', '')
+operand_pullspec = os.getenv('OPERAND_IMAGE_PULLSPEC', '')
+kube_rbac_proxy_pullspec = os.getenv('KUBE_RBAC_PROXY_IMAGE_PULLSPEC', '')
+csv = load_manifest(os.getenv('TARGET_CSV_FILE'))
+
+# Update CSV metadata and spec
+csv['metadata']['annotations']['createdAt'] = datetime_time.strftime('%Y-%m-%dT%H:%M:%S')
+csv['metadata']['annotations']['containerImage'] = operator_pullspec
+csv['metadata']['name'] = 'external-dns-operator.v{}'.format(version)
+csv['metadata']['annotations']['olm.skipRange'] = '<{}'.format(version)
+csv['spec']['version'] = version
+
+if replaces:
+    csv['spec']['replaces'] = 'external-dns-operator.v{}'.format(replaces)
+
+# Add relatedImages
+if '@sha256:' not in operator_pullspec:
+    print(f"ERROR: operator image does not contain SHA256: {operator_pullspec}")
+    exit(3)
+
+operator_sha = operator_pullspec.split('@sha256:')[1]
+annotation_image_name = f'external-dns-operator-rhel9-operator-{operator_sha}-annotation'
+csv['spec']['relatedImages'] = [
+    {'name': annotation_image_name, 'image': operator_pullspec},
+    {'name': 'operator', 'image': operator_pullspec},
+    {'name': 'kube-rbac-proxy', 'image': kube_rbac_proxy_pullspec},
+    {'name': 'external-dns', 'image': operand_pullspec}
```

Comment:
On CPaaS it was an underscore as a separator, we should stick to this for the moment:
```suggestion
    {'name': 'external_dns', 'image': operand_pullspec}
```

Can you please check with [the latest CPaaS build](https://brewweb.engineering.redhat.com/brew/buildinfo?buildID=3364237) for any other differences in the result bundle manifests please?